client/src/containers/widget/index.tsx

```diff
@@ -44,7 +52,7 @@ export default function Widget({
   };
   let ChartComponent = null;
 
-  if (currentVisualization === "single_value") {
+  if (currentVisualization === "single_value" && isWidgetData(data)) {
```

Comment:
I think this can be part of the `switch` block instead of treating it as an exception.